File: decrypt.py
from base64 import b64encode, b64decode
from Crypto.Cipher import AES
from Crypto.Protocol import KDF

KEY_SIZE_BYTES = 32;
PBKDF2_ITERATIONS  = 50000

password = ""
encryption_iv = ""
encryption_salt = "="
encryptedData = ""

def get_encryption_key(password: str, salt: str):
    #based on https://github.com/Jessecar96/SteamDesktopAuthenticator/blob/8a408f13ee24f70fffbc409cb0e050e924f4fe94/Steam%20Desktop%20Authenticator/FileEncryptor.cs#L72
    return KDF.PBKDF2(password, b64decode(encryption_salt), dkLen=KEY_SIZE_BYTES, count=PBKDF2_ITERATIONS)


def decrypt(data):
    #based on https://github.com/Jessecar96/SteamDesktopAuthenticator/blob/8a408f13ee24f70fffbc409cb0e050e924f4fe94/Steam%20Desktop%20Authenticator/FileEncryptor.cs#L87
    cipher_text = b64decode(data)
    key = get_encryption_key(password, encryption_salt)

    unpad = lambda s: s[:-ord(s[len(s) - 1:])] #replaces aes_padding
    
    aes_iv = b64decode(encryption_iv)
    aes_key = key #no idea why this is done like this but i'm replicating the src
    aes_padding = None
    aes_mode = AES.MODE_CBC
    
    aes_decrypter = AES.new(aes_key, aes_mode, aes_iv)
    cipher_text = aes_decrypter.decrypt(b64decode(data))
    return unpad(cipher_text)

def main():
    plain_text = decrypt(encryptedData)
    print(plain_text)
    
    
    
if __name__ == "__main__":
    main()

File: decrypt.js
import crypto from "crypto"

const PBKDF2_ITERATIONS = 50000; //Set to 50k to make program not unbearably slow. May increase in future.
const SALT_LENGTH = 8;
const KEY_SIZE_BYTES = 32;
const IV_LENGTH = 16;


const password = '';
const encryptedData = '';
const encryptionSalt = '';
const encryptionIv = '';

/**
 * Based on https://github.com/Jessecar96/SteamDesktopAuthenticator/blob/8a408f13ee24f70fffbc409cb0e050e924f4fe94/Steam%20Desktop%20Authenticator/FileEncryptor.cs#L72
 */
function getEncryptionKey(password, salt) {
    return new Promise(resolve => {
        crypto.pbkdf2(password, Buffer.from(salt, 'base64'), PBKDF2_ITERATIONS, KEY_SIZE_BYTES, 'sha1', ((err, derivedKey) => {
            resolve(derivedKey)
        }))
    })
}

/**
 * Based on https://github.com/Jessecar96/SteamDesktopAuthenticator/blob/8a408f13ee24f70fffbc409cb0e050e924f4fe94/Steam%20Desktop%20Authenticator/FileEncryptor.cs#L87
 *
 * @param password Your encryption password.
 * @param encryptionSalt `encryption_salt` in manifest.json
 * @param encryptionIv `encryption_iv` in manifest.json
 * @param encryptedData Content of the encrypted .maFile
 */
async function decryptData(password, encryptionSalt, encryptionIv, encryptedData) {
    const key = await getEncryptionKey(password, encryptionSalt)

    const decrypter = crypto.createDecipheriv("aes-256-cbc", key, Buffer.from(encryptionIv, 'base64'));

    return decrypter.update(encryptedData, "base64", "utf8") + decrypter.final("utf8");
}

async function main() {
    const plaintext = await decryptData(password, encryptionSalt, encryptionIv, encryptedData);
    console.log(plaintext)
}

main()
